File: Data Structures & Algorithms/append-characters-to-string-to-make-subsequence/submission-2.java
class Solution {
    public int appendCharacters(String s, String t) {
        int nextCharIndex = 0;
        for (int i = 0; i < s.length(); i++) {
            if (nextCharIndex == t.length()) {
                break;
            }
            if (s.charAt(i) == t.charAt(nextCharIndex)) {
                nextCharIndex++;
            }
        }

        if (nextCharIndex == t.length()) {
            return 0;
        }
        
        return t.length() - nextCharIndex;
    }
}

File: Data Structures & Algorithms/is-anagram/submission-2.js
class Solution {
    /**
     * @param {string} s
     * @param {string} t
     * @return {boolean}
     */
    isAnagram(s, t) {
        if (s.length != t.length) {
            return false;
        }
        const counter = new Map();
        for (let i = 0; i < s.length; i++) {
            const c = s.charAt(i);
            if (counter.has(c)) {
                counter.set(c, counter.get(c) + 1);
            } else {
                counter.set(c, 1);
            }
        }

        for (let j = 0; j < t.length; j++) {
            const c = t.charAt(j);
            if (counter.has(c)) {
                if (counter.get(c) - 1 < 0) {
                    return false;
                }
                counter.set(c, counter.get(c) - 1);
            } else {
                return false;
            }
        }
        return true;
    }
}
